index.go

```diff
@@ -633,61 +725,78 @@ border-radius: 3px;`
 		FirstChild: htmlg.Text(d.Text),
 	}
-	div := htmlg.DivClass("details",
-		code,
-	)
-	return []*html.Node{div}
+	return []*html.Node{code}
 }
 
-type commitsDetails struct {
+type commits struct {
 	Commits []*github.RepositoryCommit
 }
 
-func (d commitsDetails) Render() []*html.Node {
+func (d commits) Render() []*html.Node {
 	var nodes []*html.Node
 
 	for _, c := range d.Commits {
-		avatar := &html.Node{
-			Type: html.ElementNode, Data: atom.Img.String(),
-			Attr: []html.Attribute{
-				{Key: atom.Src.String(), Val: *c.Author.AvatarURL},
-				{Key: atom.Style.String(), Val: "width: 16px; height: 16px; vertical-align: top; margin-right: 6px;"},
-			},
-		}
-		sha := &html.Node{
-			Type: html.ElementNode, Data: atom.Code.String(),
-			FirstChild: htmlg.Text(shortSHA(*c.SHA)),
-		}
-		if c.HTMLURL != nil {
-			sha = &html.Node{
-				Type: html.ElementNode, Data: atom.A.String(),
-				Attr: []html.Attribute{
-					{Key: atom.Href.String(), Val: *c.HTMLURL},
-				},
-				FirstChild: sha,
-			}
-		}
-		message := &html.Node{
-			Type: html.ElementNode, Data: atom.Span.String(),
-			Attr: []html.Attribute{
-				{Key: atom.Style.String(), Val: "margin-left: 6px;"},
-			},
-			FirstChild: htmlg.Text(firstParagraph(*c.Commit.Message)),
-		}
-
-		div := htmlg.Div(avatar, sha, message)
+		div := htmlg.Div(commit{C: c}.Render()...)
 		div.Attr = append(div.Attr, html.Attribute{Key: atom.Style.String(), Val: "margin-top: 4px;"})
 		nodes = append(nodes, div)
 	}
 
-	div := htmlg.DivClass("details", nodes...)
-	return []*html.Node{div}
+	return nodes
+}
+
+type commit struct {
+	C     *github.RepositoryCommit
+	Short bool
+}
+
+func (c commit) Render() []*html.Node {
+	avatar := &html.Node{
+		Type: html.ElementNode, Data: atom.Img.String(),
+		Attr: []html.Attribute{
+			{Key: atom.Src.String(), Val: *c.C.Author.AvatarURL},
+			{Key: atom.Style.String(), Val: "width: 16px; height: 16px; vertical-align: top; margin-right: 4px;"},
+		},
+	}
+	sha := &html.Node{
+		Type: html.ElementNode, Data: atom.Code.String(),
+		FirstChild: htmlg.Text(shortSHA(*c.C.SHA)),
+	}
+	if c.C.HTMLURL != nil {
+		sha = &html.Node{
+			Type: html.ElementNode, Data: atom.A.String(),
+			Attr: []html.Attribute{
+				{Key: atom.Href.String(), Val: *c.C.HTMLURL},
+			},
+			FirstChild: sha,
+		}
+	}
+	message := &html.Node{
+		Type: html.ElementNode, Data: atom.Span.String(),
+		Attr: []html.Attribute{
+			{Key: atom.Style.String(), Val: "margin-left: 4px;"},
+			{Key: atom.Title.String(), Val: *c.C.Commit.Message},
+		},
+	}
+	switch c.Short {
+	case false:
+		message.AppendChild(htmlg.Text(firstParagraph(*c.C.Commit.Message)))
+	case true:
+		message.AppendChild(htmlg.Text(shortCommit(firstParagraph(*c.C.Commit.Message))))
+	}
+	return []*html.Node{avatar, sha, message}
 }
 
 func shortSHA(sha string) string {
 	return sha[:8]
 }
 
+func shortCommit(s string) string {
+	if len(s) <= 24 {
+		return s
+	}
+	return s[:23] + "…"
+}
```

Comment:
That would make it harder to adjust the 4 types of lengths I use. It's currently 200 for bodies, 36 for title, 24 for commit message, and 8 for SHAs. That corresponds to the 4 funcs.

If I used `ellipsis(s string, l int)` everywhere, it'd be harder to update those 4 categories.

I _could_ rewrite `shortCommit`, `shortBody` and `shortTitle` to use `ellipsis(s string, l int)`, but it doesn't seem to simplify the code that much:

```diff
diff --git a/index.go b/index.go
index 1ce36b1..5b7cb13 100644
--- a/index.go
+++ b/index.go
@@ -694,19 +694,9 @@ func (d imageText) Render() []*html.Node {
 	return []*html.Node{div}
 }
 
-func shortBody(s string) string {
-	if len(s) <= 200 {
-		return s
-	}
-	return s[:199] + "…"
-}
+func shortBody(s string) string { return ellipsis(s, 200) }
 
-func shortTitle(s string) string {
-	if len(s) <= 36 {
-		return s
-	}
-	return s[:35] + "…"
-}
+func shortTitle(s string) string { return ellipsis(s, 36) }
 
 type code struct {
 	Text          string
@@ -790,12 +780,7 @@ func shortSHA(sha string) string {
 	return sha[:8]
 }
 
-func shortCommit(s string) string {
-	if len(s) <= 24 {
-		return s
-	}
-	return s[:23] + "…"
-}
+func shortCommit(s string) string { return ellipsis(s, 24) }
 
 // firstParagraph returns the first paragraph of text s.
 func firstParagraph(s string) string {
@@ -854,3 +839,10 @@ func (d pages) Render() []*html.Node {
 
 	return nodes
 }
+
+func ellipsis(s string, l int) string {
+	if len(s) <= l {
+		return s
+	}
+	return s[:l-1] + "…"
+}
```

There's only 3 funcs, they're very short and easy to read, modify. And I expect they may change or go away soon. That's why I don't think trying to generalize them right now would buy a lot.

Thanks for the suggestion!